src/test/registration_test.go

```diff
@@ -196,8 +201,12 @@ func TestRegistration(t *testing.T) {
 	// receive NAS Security Mode Command Msg
 	n, err = conn.Read(recvMsg)
 	assert.Nil(t, err)
-	_, err = ngap.Decoder(recvMsg[:n])
+	ngapPdu, err = ngap.Decoder(recvMsg[:n])
 	assert.Nil(t, err)
+	assert.NotNil(t, ngapPdu)
+	nasPdu = test.GetNasPdu(ue, ngapPdu.InitiatingMessage.Value.DownlinkNASTransport)
+	assert.NotNil(t, nasPdu)
+	assert.True(t, nasPdu.GmmHeader.GetMessageType() == nas.MsgTypeSecurityModeCommand, "No Security Mode Command received. Message: "+strconv.Itoa(int(nasPdu.GmmHeader.GetMessageType())))
```

Comment:
This potentially crashes if the nasPdu doesn't contain a Gmm message, we'd have to check the EPD for GMM/GSM, too.

Reply:
Sure, we try to check this problems. Thanks for the report!